unittests/APITests/StringUtils.cpp

```diff
@@ -0,0 +1,91 @@
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/generators/catch_generators.hpp>
+#include <Common/StringUtil.h>
+
+#define trim_test_type(trim_type, in, out) \
+  do {                                     \
+    fextl::string str = in;                \
+    FEX::StringUtil::trim_type(str);       \
+    CHECK(str == out);                     \
+  } while (0)
```

Comment:
Don't use macros when a function would do. As usual macros don't combine well with things, in this case they break proper test reporting.

Reply:
```cpp
std::string get_Ntrimmed_string(std::string str) \
    Ntrimmed(str);
    return str;
}
```

Reply:
didit